### PR TITLE
Subnet De-registration Chart

### PR DESCRIPTION
**Pull request by TopETH**

### CODE REVIEW

File: src/components/subnets/NeuronDeregistrationChart.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	const incentive = useMemo(() => {
+		if (!neuronDeregistrations.data) return [];
+		return neuronDeregistrations.data.map(({ incentive }) => incentive / 65535);
+	}, [neuronDeregistrations]);
+	const emission = useMemo(() => {
+		if (!neuronDeregistrations.data) return [];
+		return neuronDeregistrations.data.map(({ emission }) =>
+			rawAmountToDecimal(emission.toString()).toNumber()
+		);
+	}, [neuronDeregistrations]);
```
Comment (cuteolaf) [inline src/components/subnets/NeuronDeregistrationChart.tsx:64]:
```suggestion
	const timestamps = useMemo(() => {
		const { data } = neuronDeregistrations;
		if (!data) return [];
		return data.map((x: NeuronDeregistration) => x.timestamp);
	}, [neuronDeregistrations]);
	
	const [minIncentive, maxIncentive] = useMemo(() => {
		const { data } = neuronDeregistrations;
		if (!data) return [0, 0];

		const dat = data.map(({ incentive }) => incentive / 65535);
		return [Math.min(...dat), Math.max(...dat)];
	}, [neuronDeregistrations]);
	
	const [minEmission, maxEmission] = useMemo(() => {
		const { data } = neuronDeregistrations;
		if (!data) return [0, 0];

		const dat = data.map(({ emission }) =>
			rawAmountToDecimal(emission.toString()).toNumber()
		);
		return [Math.min(...dat), Math.max(...dat)];
	}, [neuronDeregistrations]);
	
	const incentives = useMemo(() => {
		const { data } = neuronDeregistrations;
		if (!data) return [];
		return data.map(({ incentive }) => incentive / 65535);
	}, [neuronDeregistrations]);
	
	const emissions = useMemo(() => {
		const { data } = neuronDeregistrations;
		if (!data) return [];
		return data.map(({ emission }) =>rawAmountToDecimal(emission.toString()).toNumber());
	}, [neuronDeregistrations]);
```

File: src/components/subnets/NeuronDeregistrationChart.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+			<img src={LoadingSpinner} />
+		</div>
+	) : (
+		<Chart
+			height={400}
+			series={[
+				{
+					name: "Incentive",
+					type: "area",
+					data: incentive,
```
Comment (cuteolaf) [inline src/components/subnets/NeuronDeregistrationChart.tsx:77]:
```suggestion
					data: incentives,
```

File: src/components/subnets/NeuronDeregistrationChart.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+				},
+				{
+					name: `Emission (${NETWORK_CONFIG.currency})`,
+					type: "area",
+					data: emission,
```
Comment (cuteolaf) [inline src/components/subnets/NeuronDeregistrationChart.tsx:82]:
```suggestion
					data: emissions,
```

File: src/hooks/useNeuronDeregistrations.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			const result: NeuronDeregistration[] = [];
+			while (!finished) {
+				const deregistration: NeuronDeregistrationPaginatedResponse =
+					await getNeuronDeregistration(
+						{
+							netUid: {
+								equalTo: parseInt(id),
+							},
+							timestamp: {
+								greaterThan: new Date(from).toISOString().substring(0, 19),
```
Comment (cuteolaf) [inline src/hooks/useNeuronDeregistrations.ts:38]: ```const from = new Date(now - 7 * 24 * 60 * 60 * 1000).toISOString();```

and

```new Date(from).toISOString()```
?

File: src/model/subnet.ts [hunk 5]
```diff
@@ -140,6 +140,26 @@ export type NeuronRegCostHistoryResponse = {
 	data: NeuronRegCostHistory[];
 };
 
+export type NeuronDeregistration = {
+	height: bigint;
```
Comment (cuteolaf) [inline src/model/subnet.ts:144]:
```suggestion
	height: number;
```

height is of type `number`, not `bigint`

**Review by cuteolaf — CHANGES_REQUESTED** [review]

**Review by cuteolaf — APPROVED** [review]

**Review by mogmachine — APPROVED** [review]

**Pull request merged**